src/mcts/search.cc

```diff
@@ -189,13 +189,17 @@ int64_t Search::GetTimeToDeadline() const {
 
 namespace {
 inline float GetFpu(const SearchParams& params, Node* node, bool is_root_node) {
+  // Use root FPU behavior unless it's "same"
+  if (is_root_node) {
+    if (params.GetFpuAbsoluteAtRoot()) return params.GetFpuValueAtRoot();
+    if (params.GetFpuReductionAtRoot())
+      return -node->GetQ() -
+             params.GetFpuValueAtRoot() * std::sqrt(node->GetVisitedPolicy());
+  }
   return params.GetFpuAbsolute()
```

Comment:
Not sure if you wanted this kept as a single return statement (for inline?) -- if so, the logic could be:
```c
return is_root_node && (absoluteRoot || reductionRoot)
  ? absoluteRoot ? valueRoot : -Q - valueRoot*visited
  : absolute
    ? value
    : -Q - value*visited
```

Reply:
Alternatively, if we make the param getters be absoluteRoot and sameRoot (instead of adding reductionRoot):
```c
return is_root_node && !sameRoot
  ? absoluteRoot ? valueRoot : -Q - valueRoot*visited
  ? absolute ? value : -Q - value*visited
```

Reply:
Maybe parse everything in `Params` class and make `GetFpuAbsolute` and `GetFpuValue` have parameter there, and leave here something like:
```cpp
  const auto value = params.GetFpuValue(is_root_node);
  return params.GetFpuAbsolute(is_root_node) ? value : (-Q - value * std::sqrt(node->GetVisitedPolicy()));
```

Reply:
Interesting. I believe this would make it easier for "same" to allow differing values at root. Should that be allowed? Or keep the PR's behavior of "same" == don't treat root differently?

```
strategy = reduction
value = 1.2
strategy-at-root = same (i.e., reduction)
value-at-root = 0

or

strategy = absolute
value = -1
strategy-at-root = same (i.e., absolute)
value-at-root = 1
```

Reply:
I was thinking that if value is `same` then `Params` initialized both root's strategy and value from ones of non-root.
Otherwise is user wants to specify `--fpu-value=1.23` and not have special behaviour to root, they willl also have to specify `--fpu-root-value=1.23`.

Reply:
Okay, I made the Params constructor check for "same" to initialize to the other values.